noir-projects/noir-contracts/contracts/nft_contract/src/main.nr

```diff
@@ -212,19 +209,28 @@ contract NFT {
         // is zero because the slot is the x-coordinate of the hiding point and hence we could only overwrite
         // the value in the slot with the same value. This makes usage of the `unsafe` method safe.
         NFT::at(context.this_address())
-            ._store_point_in_transient_storage_unsafe(
+            ._store_payload_in_transient_storage_unsafe(
                 hiding_point_slot,
                 note_setup_payload.hiding_point,
+                setup_log,
             )
             .enqueue(context);
 
         hiding_point_slot
     }
 
+    // TODO(#9375): Having to define the note log length here is very unfortunate as it's basically impossible for users
+    // to derive manually. This will however go away once we have a real transient storage. We cannot use generics here
+    // as it is an entrypoint function.
```

Comment:
This ^ is what I mentioned in scrum.

Reply:
Why does it go away with transient storage?

Reply:
Clarified the comment in https://github.com/AztecProtocol/aztec-packages/pull/9356/commits/8f9458cac6b00817f13f7fcdda6115334046dd74

> Having to define the note log length here is very unfortunate as it's basically impossible for users to derive manually. This will however go away once we have a real transient storage since we will not need the public call and instead we would do something like `context.transient_storage_write(slot, payload)` and that will allow us to use generics and hence user will not need to define it explicitly. We cannot use generics here as it is an entrypoint function.